atc/engine/engine.go

```diff
@@ -234,6 +234,11 @@ func (b *engineBuild) Run(logger lager.Logger) {
 
 	case err = <-done:
 		logger.Debug("engine-build-done")
+		if err != nil {
+			if _, ok := err.(exec.Retriable); ok {
+				return
```

Comment:
This is pretty interesting. So the idea is to just bail on the build plan for now and allow the build tracker to pick it back up?

Don't really have any preference either way at the moment - it should be fine, and reinforces the notion that builds are idempotent and can be re-attached. It might make it difficult to implement things like "retry up to N times" though. 🤔

Reply:
I like it! 

If we wanted to do "retry up to N times" or some type of backoff, it might make sense to do that in the build tracker anyway?

Reply:
@vito @pivotal-jwinters please take a look at my second commit. In the second commit, I implemented a logic of only rerun necessary steps. With that, I just don't return to tracker at all, once a build fails, the engine retries it immediately, so that it holds the original `run_state`, from `run_state` it knows previous volumes.

Reply:
Per https://github.com/concourse/concourse/pull/5192#issuecomment-624827620, nothing to do with this comment, thus resolving.

Reply:
Per @vito, nothing to address for this comment, thus closing.